src/Hosting/Hosting/src/Internal/HostingMetrics.cs

```diff
@@ -0,0 +1,174 @@
+        if (exception != null)
+        {
+            tags.Add("exception-name", exception.GetType().Name);
+        }
+        if (customTags != null)
+        {
+            for (var i = 0; i < customTags.Count; i++)
+            {
+                tags.Add(customTags[i]);
+            }
+        }
+
+        var duration = Stopwatch.GetElapsedTime(startTimestamp, currentTimestamp);
+        _requestDuration.Record(duration.TotalMilliseconds, tags);
+    }
+
+    public void Dispose()
+    {
+        _meter.Dispose();
+    }
+
+    public bool IsEnabled() => _currentRequestsCounter.Enabled || _requestDuration.Enabled;
+
+    private static TagList CreateRequestTags(string scheme, string method, HostString host)
+    {
+        var tags = new TagList();
+        tags.Add("scheme", scheme);
+        tags.Add("method", method);
+        if (host.HasValue)
+        {
+            tags.Add("host", host.Host);
+            if (host.Port is not null && host.Port != 80 && host.Port != 443)
+            {
+                tags.Add("port", host.Port);
+            }
+        }
+
+        return tags;
+    }
+
+    // Status Codes listed at http://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml
+    private static readonly Dictionary<int, object> BoxedStatusCodes = new()
+    {
+        { 100, 100 },
+        { 101, 101 },
+        { 102, 102 },
+
+        { 200, 200 },
+        { 201, 201 },
+        { 202, 202 },
+        { 203, 203 },
+        { 204, 204 },
+        { 205, 205 },
+        { 206, 206 },
+        { 207, 207 },
+        { 208, 208 },
+        { 226, 226 },
+
+        { 300, 300 },
+        { 301, 301 },
+        { 302, 302 },
+        { 303, 303 },
+        { 304, 304 },
+        { 305, 305 },
+        { 306, 306 },
+        { 307, 307 },
+        { 308, 308 },
+
+        { 400, 400 },
+        { 401, 401 },
+        { 402, 402 },
+        { 403, 403 },
+        { 404, 404 },
+        { 405, 405 },
+        { 406, 406 },
+        { 407, 407 },
+        { 408, 408 },
+        { 409, 409 },
+        { 410, 410 },
+        { 411, 411 },
+        { 412, 412 },
+        { 413, 413 },
+        { 414, 414 },
+        { 415, 415 },
+        { 416, 416 },
+        { 417, 417 },
+        { 418, 418 },
+        { 419, 419 },
+        { 421, 421 },
+        { 422, 422 },
+        { 423, 423 },
+        { 424, 424 },
+        { 426, 426 },
+        { 428, 428 },
+        { 429, 429 },
+        { 431, 431 },
+        { 451, 451 },
+        { 499, 499 },
+
+        { 500, 500 },
+        { 501, 501 },
+        { 502, 502 },
+        { 503, 503 },
+        { 504, 504 },
+        { 505, 505 },
+        { 506, 506 },
+        { 507, 507 },
+        { 508, 508 },
+        { 510, 510 },
+        { 511, 511 }
+    };
+
+    private static object GetBoxedStatusCode(int statusCode)
+    {
+        if (BoxedStatusCodes.TryGetValue(statusCode, out var result))
```

Comment:
I chose a dictionary because an array would be 500 items in size, along with all those boxed integers. A dictionary can be sparse.

I can change it to a FrozenDictionary.

Reply:
This is probably a very "micro" tweak, not sure how much real impact are we going to get and if the extra memory is worthy.

It seems @CodeBlanch decided to use array (normally he has solid reasons) https://github.com/CodeBlanch/opentelemetry-dotnet/blob/d5aba5965d161f3f68adc52ab61ad64704cead03/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/TelemetryHelper.cs#L25.